ui/app/routes/vault/cluster/access/method/section.js

```diff
@@ -1,22 +1,66 @@
-/* eslint-disable prettier/prettier */
-import { set } from '@ember/object';
 import { inject as service } from '@ember/service';
 import Route from '@ember/routing/route';
-import DS from 'ember-data';
 
 export default Route.extend({
   wizard: service(),
+  pathHelp: service('path-help'),
 
-  model(params) {
-    const { section_name: section } = params;
-    if (section !== 'configuration') {
-      const error = new DS.AdapterError();
-      set(error, 'httpStatus', 404);
-      throw error;
+  modelType(backendType, section) {
+    const MODELS = {
+      'aws-client': 'auth-config/aws/client',
+      'aws-identity-whitelist': 'auth-config/aws/identity-whitelist',
+      'aws-roletag-blacklist': 'auth-config/aws/roletag-blacklist',
+      'azure-configuration': 'auth-config/azure',
+      'github-configuration': 'auth-config/github',
+      'gcp-configuration': 'auth-config/gcp',
+      'jwt-configuration': 'auth-config/jwt',
+      'oidc-configuration': 'auth-config/oidc',
+      'kubernetes-configuration': 'auth-config/kubernetes',
+      'ldap-configuration': 'auth-config/ldap',
+      'okta-configuration': 'auth-config/okta',
+      'radius-configuration': 'auth-config/radius',
+    };
+    return MODELS[`${backendType}-${section}`];
+  },
+
+  beforeModel() {
+    const { apiPath, method, type } = this.getMethodAndModelInfo();
+    let modelType = this.modelType(type, 'configuration');
+    if (modelType) {
+      return this.pathHelp.getNewModel(modelType, method, apiPath);
     }
-    let backend = this.modelFor('vault.cluster.access.method');
+  },
+
+  getMethodAndModelInfo() {
+    const { path: method } = this.paramsFor('vault.cluster.access.method');
+    const methodModel = this.modelFor('vault.cluster.access.method');
+    const { apiPath, type } = methodModel;
+    return { apiPath, type, method };
+  },
+
+  model() {
+    const backend = this.modelFor('vault.cluster.access.method');
+    const modelType = this.modelType(backend.type, 'configuration');
     this.wizard.transitionFeatureMachine(this.wizard.featureState, 'DETAILS', backend.type);
-    return backend;
+
+    if (!modelType) {
+      return backend; //tune options
+    }
+
+    return this.store
+      .findRecord(modelType, backend.id)
+      .then(methodConfig => {
+        methodConfig.set('backend', backend);
```

Comment:
This makes `model` potentially two different types - if it's not got an authConfig, the model is the auth-method, if it does, the model is the auth config.  Can we turn this around so that we're adding to the hasMany on the backend model and returning it?

If we do that we can also simplify this a little as we wouldn't need to `findRecord` - we could each over the authConfigs and just call `get` on it which triggers `findRecord`.

So it'd be something like `backend.get('authConfigs').pushObject(authConfig)` first and then we could loop and call get, then return the backend.